sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1735,6 +1735,12 @@ def task_worker(self, stopping_event, sfp_error_event):
                                     media_settings_parser.notify_media_setting(logical_port, transceiver_dict, self.xcvr_table_helper, self.port_mapping)
                                     transceiver_dict.clear()
                             elif value == sfp_status_helper.SFP_STATUS_REMOVED:
+                                # Remove the SFP API object for this physical port
+                                try:
```

Comment:
@jianyuewu need to make sure the API object held by other thread can become invalid. Did you test the scenario?

Reply:
![image](https://github.com/user-attachments/assets/f75f0d35-17ac-4bc2-a035-0341c0ff4bec)